x/auth/middleware/testutil_test.go

```diff
@@ -0,0 +1,142 @@
+package middleware_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	xauthsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+)
+
+// testAccount represents an account used in the tests in x/auth/ante.
+type testAccount struct {
+	acc  authtypes.AccountI
+	priv cryptotypes.PrivKey
+}
+
+// MWTestSuite is a test suite to be used with ante handler tests.
+type MWTestSuite struct {
```

Comment:
This file is a copy-paste from `x/auth/ante`'s test suite. In Part 2, we'll remove `x/auth/ante`.